pkg/metadata/scheduler.go

```diff
@@ -101,13 +105,25 @@ func (c *Scheduler) AddCollector(name string, interval time.Duration) error {
 	}()
 	c.collectors[name] = sc
 
+	if withInit, ok := p.(CollectorWithInit); ok {
+		withInit.Init()
+	}
+
 	return nil
 }
 
-// SendNow runs a collector immediately and resets its ticker.
-// Does nothing if the Scheduler has been stopped, since the
-// goroutine that listens on the Timer will not be running.
-func (c *Scheduler) SendNow(name string) {
+// IsScheduled returns wether a given Collector has been added to this Scheduler
+func (c *Scheduler) IsScheduled(name string) bool {
+	_, found := c.collectors[name]
+	return found
+}
+
+// SendNow runs a collector manually, on demand. The delay parameter can be set
+// to 0 to run it immediately, or to a duration after which the collector will
+// run. The runs at regular intervals of the collector will be resumed *after*
+// this manual run. Calling SendNow on a stopped Scheduler does nothing, since
+// the goroutine that waits on the Timer will not be running.
+func (c *Scheduler) SendNow(name string, delay time.Duration) {
```

Comment:
Could we rename this, I don't think the name describes the function clearly. `TriggerAndResetCollectorTimer` maybe ?

Reply:
It was hard for me to come up with a good name. `TriggerAndResetCollectorTimer` is pretty descriptive, I'll change it.